articles/node-vs-nuxt/index.md

```diff
@@ -0,0 +1,160 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/node-vs-nuxt/
+title: Node vs Nuxt - The Key Differences
+description: .
+author: saiharsha-balasubramaniam
+date: 2020-09-06T00:00:00-12:00
+topics: [Node.js]
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/node-eslint/hero.jpg
+    alt: linters example image ESLint
+---
+
+A while ago, traditional server side rendering used to be the norm. All of the HTML is dynamically rendered on the server and sent to the client. Back in those days, websites weren't user-friendly and native.
+
+Today's web consists of single-page applications and universal web-apps. Let us look at the differences between Node.js and Nuxt which are two important technologies used for the web.
+
+<!--more-->
+
+## Table of Contents
+
+- [Why is Node.js Needed](#why-is-node.js-needed)
+- [Why is Nuxt Needed](#why-is-nuxt-needed)
+- [Node.js-vs-Nuxt](#node.js-vs-nuxt)
+- [Create a Node.js App](#create-a-node.js-app)
+- [Create a Nuxt App](#create-a-nuxt-app)
+- [Further Reading](#further-reading)
+
+### Why is Node.js Needed
+
+JavaScript is a powerful object-oriented programming language that runs within a browser. To harness the power of JS on the server-side, Node.js was created.
+
+Node.js is a server-side JavaScript runtime environment. It is based on the Google V8 engine. Today, it is one of the most popular server-side languages for the web.
+
+Node.js is single threaded and event driven. Node.js is not a framework, but it is an environment. There are many frameworks that are built on top of Node.js, namely Express and Koa. It is also preferred because front-end developers can easily jump into backend development, because they would already know JavaScript.
+
+### Why is Nuxt Needed
+
+Since traditional server-side rendering fails to provide a smooth user experience, client-side rendering became the new standard for web applications. Client-side rendering involves a single HTML file being sent to the client. Now, the entire JavaScript is downloaded and runs on the client to handle interactivity.
+
+This had a few drawbacks with Search Engine Optimization being the most crucial one. The SEO parser isn't able to parse all the DOM elements due to which SEO takes a hit. Here, pages are rendered both on the client and the server-side. This improves SEO and the initial page load times.
```

Comment:
You should give an example or link to a graphic that shows the difference between server-side rendering and client-side rendering.